libs/rcm/include/celix_rcm_err.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CELIX_CELIX_RCM_ERR_H
+#define CELIX_CELIX_RCM_ERR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file celix_rcm_err.h
+ * @brief The rcm error handling functions.
+ * @thread_safety Thread safe.
+ */
+
+/**
+ * @brief Returns the last error message from the current thread.
+ * The error message must be freed by the caller.
+ * @returnval NULL if no error message is available.
+ */
+char* celix_rcmErr_popLastError();
```

Comment:
I also think this is not a rcm thing. I just added it to rcm to kick of the discussion. 

IMO this could be a separate lib (potentially depending on utils) or part of utils self. 

I think the API still needs some discussion and i not sure if the setup/teardown should be done with `__attribute__((constructor/destructor))` or with a set of `err_init()/err_deinit()` calls.

Reply:
> i not sure if the setup/teardown should be done with __attribute__((constructor/destructor)) or with a set of err_init()/err_deinit() calls.

I didn't see obvious downsides of `__attribute__((constructor/destructor))`. 
`pthread_once` could alse be an alternative.
Battle-tested ERR can also serve as our guide. 
I will have a good read of its implementation this weekend.

Reply:
> Another thing concerning me is dynamic allocation.
> ERR also supports `celix_rcmErr_pushf` and uses dynamic memory allocation in that case.
> But I remember in lots of case constant error string is used, and thus without malloc. But I might be wrong.

Agree, every where now a fprintf is needed to handle a possible ENOMEM should be avoided. The ERR api suggest that no free of the error string is needed, so I suspect something like a thread specific ring buffer + entry prt. 
But indeed good to check the implementation.